### PR TITLE
feat: add next reminder date to stayintouch

### LINKED ISSUE #5158: Option to display date for next "Stay in touch" reminder
It would be useful if there was an option to display the date of next reminder.  This could be done one of two ways:

1. Text in brackets next to the reminder:  _"Stay in touch every 40 days (next due 05/05/2021)"_
2. Using the `<abbr>` tag so that you can see it by hovering over the frequency

The first approach is simpler, and works everywhere, but at the expense of a little clutter.  The second approach is neater, but not easily discoverable and useless on mobile.

### PR DESCRIPTION
**Pull request by MarkusDick**
Fix #5158 

The screenshot of this feature:
next_reminder_stay_in_touch

I had to change the ContactController stayInTouch function because I needed the new calculated next reminder date when I update the frequency of the reminder.
Otherwise I'ld have needed to make a second request to the API.

#### Before submitting the PR
- [x] Read the [CONTRIBUTING document](https://github.com/monicahq/monica/blob/master/CONTRIBUTING.md) before submitting your PR.
- [x] If the PR is related to an issue or fix one, don't forget to indicate it.

### General checks
- [x] Make sure that the change you propose is the smallest possible.
- [x] The name of the PR should follow the [conventional commits guideline](https://github.com/monicahq/monica/blob/master/docs/contribute/readme.md#conventional-commits) that the project follows.

### Front-end changes
- [x] If you change the UI, make sure to ask repositories administrators first about your changes by pinging djaiss or asbiin in this PR. (pinged in issue)
- [x] Screenshots are included if the PR changes the UI.

### CODE REVIEW

**Review by asbiin — COMMENTED** [review]
I think adding the whole part "(next due ...)"  is quite long. Wouldn't be better to just add a tooltip-like when the mouse is over the text?

**Comment by MarkusDick** [comment]
That's a great idea. I'll try it and update that PR. I'll convert it to draft.

**Comment by MarkusDick** [comment]
Should the cursor be a pointer and the text "Stay in touch.." be underline dotted?

**Comment by mgrhm** [comment]
> Should the cursor be a pointer and the text "Stay in touch.." be underline dotted?

Does a tool-tip work on mobile?  About half the time I use Monica, it's on my phone.  Almost all the times I want to check something quickly, it's using my phone.  (And is a tool-tip different than [using the `<abbr>` tag](https://github.com/monicahq/monica/issues/5158#issue-874506019)?)

Dotted underline is very common for the `<abbr>` tag, so many (most?) web users should know it means to hover on that for more information.

Not sure if this is possible, but my preference would be for an option to switch between the two.  Maybe make the tool-tip the default, but have an option to switch to the "(next due 08/09/2021)"?  While the full version is clumsier, it's more accessible.

**Comment by MarkusDick** [comment]
> Does a tool-tip work on mobile?

Yes it does! :)

Here are screenshots:
Desktop:
tooltip-desktop

Mobile:
tooltip-mobile

I'll push it in the evening. Are there any suggestions for improvement?

**Comment by mgrhm** [comment]
Looks great, and glad it works on mobile!

Is there a trailing space at the end of “days” inside the tooltip tag? It looks like there might be on those screenshots.

The only other thing I’d do is stop it from [wrapping across lines](https://github.com/monicahq/monica/pull/5304). You can do that using the `nowrap-link` class.

**Comment by mgrhm** [comment]
One other thought: since it’s a tooltip rather than inline, I’d prefer displaying the full date rather than the abbreviated three-letter month.

**Comment by MarkusDick** [comment]
> I’d prefer displaying the full date rather than the abbreviated three-letter month.

Should it look like `09/23/2021` or `September 23, 2021`? @mgrhm 

> The only other thing I’d do is stop it from wrapping across lines. You can do that using the nowrap-link class.

> Is there a trailing space at the end of “days” inside the tooltip tag? It looks like there might be on those screenshots.

Both added / fixed. 👍

**Comment by mgrhm** [comment]
> Should it look like 09/23/2021 or September 23, 2021?

I’d choose the latter. “September 23, 2021” is easier to read than “09/23/2021”.

**Comment by MarkusDick** [comment]
I've added custom css for the dashed border because https://tachyons.io/docs/themes/borders/ is a lil bit buggy:


with the css classes `bb b--dashed` it looks like:
buggy

It's a known issue: https://github.com/tachyons-css/tachyons/issues/574

Now the finished look is:
Desktop:
tooltip-desktop

Mobile:


@asbiin Can u have a look at it? I don't know why the StyleCI is failing.

**Comment by asbiin** [comment]
StyleCI is failing upstream, I'll fix it.

File: resources/js/components/people/StayInTouch.vue [hunk 1]
```diff
@@ -23,7 +27,7 @@
 
     <!-- Contact has a frequency set -->
     <div v-else class="di">
-      <span>
+      <span v-tooltip.bottom="$t('people.stay_in_touch_next_date', { date: formatDate(next_trigger_date)} )" class="bb dashed dib pointer nowrap-link">
```
Comment (asbiin) [inline resources/js/components/people/StayInTouch.vue:30]: consistency change
```suggestion
      <span v-tooltip.bottom="$t('people.stay_in_touch_next_date', { date: formatDate(next_trigger_date) })" class="bb dashed dib pointer nowrap-link">
```

**Review by asbiin — CHANGES_REQUESTED** [review]
Great job! Thank you.

**Review by asbiin — APPROVED** [review]

**Comment by asbiin** [comment]
Thank you @MarkusDick

**Pull request merged**